tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = f"Error ENOENT: osd.{new_osd} does not exist"
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.error(f"An error occurred but was expected: {e}")
+                    log.debug(
+                        "Could not change the primary with non existent OSD. Pass"
+                    )
+
+                # Trying to set a OSD which is not present in the acting set. -ve test
+                osd_list = []
+                for node in ceph_cluster:
+                    if node.role == "osd":
+                        node_osds = rados_obj.collect_osd_daemon_ids(node)
+                        osd_list = osd_list + node_osds
+
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                for item in osd_list:
+                    if item not in acting_set:
+                        new_osd = item
+
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = (
+                    f"Error EINVAL: osd.{new_osd} is not in acting set for pg {pgid}"
+                )
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.error(f"An error occurred but was expected: {e}")
+                    log.debug(
+                        "Could not change the PG set with OSD which is not present in the acting set.. Pass"
+                    )
+
+                # Trying to change the primary with secondary. Should be possible
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                new_osd = acting_set[1]
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = f"change primary for pg {pgid} to osd.{new_osd}"
+                out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                if error_str not in err.strip():
+                    log.error(
+                        "Correct error string not seen in stderr stream"
```

Comment:
Maybe rephrase it to "Correct pass string not seen in STDERR"

Reply:
Done.